pkgs/development/tools/build-managers/cmake/setup-hook.sh

```diff
@@ -49,7 +49,13 @@ cmakeConfigurePhase() {
 
     # Avoid cmake resetting the rpath of binaries, on make install
     # And build always Release, to ensure optimisation flags
-    cmakeFlags="-DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON $cmakeFlags"
+    cmakeFlags="-DCMAKE_SKIP_BUILD_RPATH=ON $cmakeFlags"
+
+    if [ -n "$cmakeBuildType" ]; then
+        cmakeFlags="-DCMAKE_BUILD_TYPE=$cmakeBuildType $cmakeFlags"
+    elif [ -z "dontAddBuildType" ]; then
```

Comment:
This should read `"$dontAddBuildType`, it's missing the `$`.

A little bikeshedding here: could we call this `cmakeDontSetBuildType` instead? The `cmake` prefix matches the other CMake-specific variables. I'm less particular on `Set` instead of `Add`, but it sounds like a better verb to me.